### DISCUSSION TITLE
Failing to load `opensnitch-dns` BPF module

### DISCUSSION THREAD

**Discussion by vobst** [evilsocket/opensnitch#1210] [General]
I get the following error when trying to start `opensnitchd` with the ebpf backend:

```
[2024-10-17 09:03:22]  DBG  [eBPF] trying to load /etc/opensnitchd/opensnitch-dns.o
[2024-10-17 09:03:22]  ERR  [eBPF DNS]:
unable to load eBPF module (opensnitch-dns.o). Your kernel version (6.11.3-1-vpao) might not be compatible.
If this error persists, change process monitor method to 'proc'
[2024-10-17 09:03:22]  WAR  EBPF-DNS: Unable to attach ebpf listener:
unable to load eBPF module (opensnitch-dns.o). Your kernel version (6.11.3-1-vpao) might not be compatible.
If this error persists, change process monitor method to 'proc'
```

The other programs load fine and I can see them with `bpftool`. I use a self-compiled kernel so I suspect I may be missing some config (I do have CONFIG_UPROBES=y and CONFIG_UPROBE_EVENTS=y and all the other ones you mention). Is there a way to get more verbose debugging output about why the loading fails? Currently I'm not even sure if it fails in user-space or in some syscall.

**Comment by vobst** [comment]
Magically solved itself ...

**Reply by gustavo-iniguez-goya** [reply]
In this particular case with the ebpf DNS module, launching the daemon manually from the terminal could have printed a little more information.

```bash
~ $ sudo systemctl stop opensnitch
~ $ sudo su
~ # opensnitchd
found /lib/x86_64-linux-gnu/libc.so.6
```

Also, `opensnitchd -check-requirements` should reveal if the kernel has all the needed features.
